security/self-assessment.md

```diff
@@ -0,0 +1,228 @@
+
+- **Advanced Data Processing or Analytics:**  
+  OpenEBS does not include integrated data processing, advanced query capabilities, or native analytics functions. Its role is to provide reliable and secure storage services, leaving data processing to external systems.
+
+## Self-assessment use
+
+This self-assessment is created by the OpenEBS team to perform an internal analysis of the project’s security. It is not intended to provide a security audit of OpenEBS, or function as an independent assessment or attestation of OpenEBS’s security health.
+
+This document serves to provide OpenEBS users with an initial understanding of OpenEBS’s security, where to find existing security documentation, OpenEBS plans for security, and general overview of OpenEBS security practices, both for development of OpenEBS as well as security of OpenEBS.
+
+This document provides the CNCF TAG-Security with an initial understanding of OpenEBS to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when OpenEBS seeks graduation and is preparing for a security audit.
+
+## Security Functions and Features
+
+- **RBAC & Access Control:**  
+  - Enforces Kubernetes RBAC for secure storage operations.
+  - Uses CSI volume modes and NVMe Reservations to guarantee exclusive volume access.
+
+- **Secure Communication:**  
+  - Utilizes secure HTTP/gRPC channels between control-plane and node components.
+  - Designed for integration with external encryption solutions.
+
+- **Isolation & Privilege Separation:**  
+  - Runs node-level plugins in tightly controlled, privileged containers.
+  - Separates control-plane and node functions to minimize risk exposure.
+
+- **Data Integrity & Resilience:**  
+  - Implements secure volume lifecycle management (provisioning, expansion, deletion, snapshots).
+  - Supports replication and failover to ensure high availability and durability.
+
+- **Monitoring & Telemetry:**  
+  - Exposes metrics via Prometheus-compatible endpoints.
+  - Provides audit and telemetry data to support ongoing security evaluation.
+
+- **Kubernetes Ecosystem Integration:**  
+  - Leverages native Kubernetes APIs for consistent security policy enforcement.
+  - Modular design enables easy integration with additional security tools.
+
+## Secure development practices
+
+- **Peer Review & Static Analysis:**  
+  All code changes undergo mandatory peer reviews and automated static analysis to catch security issues early.
+
+- **Comprehensive Testing:**  
+  Extensive unit and integration tests validate functionality and secure behavior.
+
+- **Automated CI/CD Pipelines:**  
+  Secure pipelines enforce coding standards, run vulnerability scans, and perform dependency checks before deployment.
```

Comment:
run vulnerability scans --> we don't do that on CI/CD, rather it's done by the dependabot

Reply:
We do it on linux-utils. But yes, we don't directly do vulnerability scans.

Reply:
Removed the run vulnerability scans bit, as it's not a staple of our CI worflows.